exonum/src/blockchain/schema.rs

```diff
@@ -122,10 +122,15 @@ impl<T: Access> Schema<T> {
 
     /// Returns a record of errors that occurred during execution of a particular block.
     ///
-    /// This method can be used to retrieve a proof that execution of a certain transaction
-    /// ended up with a particular status. Since the number of transaction in a block is
-    /// mentioned in the block header, a proof of absence of an error for a transaction
-    /// with a particular index means that it was executed successfully.
+    /// This method can be used to build a proof that execution of a certain transaction
+    /// ended up with a particular status.
+    /// For an execution that resulted in an error, this will be an usual proof of existence.
+    /// If transaction was executed successfully, such a proof will be a proof of absence.
+    /// Since the number of transactions in a block is mentioned in the block header, the user
+    /// will be able to distinguish absence of error (meaning successful execution) from
+    /// absence of transaction with such an index: if index is less than amount of transactions
```

Comment:
I'd note that by the key itself it is not possible to tell if it belongs to the execution of a transaction with a certain message hash. My understanding is that a proper proof must first prove the tx inclusion in a block at a certain location, then — the call result (using an already proved location).

I had briefly described it here:
* https://github.com/exonum/exonum-java-binding/blob/d47524e388340b64d5e12366a658ac21d83ce931/exonum-java-binding/core/src/main/java/com/exonum/binding/core/blockchain/Blockchain.java#L302-L309
* https://github.com/exonum/exonum-java-binding/blob/d47524e388340b64d5e12366a658ac21d83ce931/exonum-java-binding/core/src/main/java/com/exonum/binding/core/blockchain/Blockchain.java#L81-L86

Reply:
I.e., one shall not check the index against the number of transaction in a block, but _construct_ a key by using an already proved location.

Reply:
I added a note that this kind of proof should not be considered a proof of inclusion. Are you OK with such a statement?

Reply:
Oh snap! It seems like I forgot to push this change. Redirected the request (https://github.com/exonum/exonum/pull/1724#discussion_r369362951).